contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,1029 @@
+        }
+
+        /**
+         * Convenience method for {@link #CFArrayGetValueAtIndex} on this object
+         *
+         * @param idx
+         *            The index of the value to retrieve.
+         * @return The value at the {@code idx} index.
+         */
+        public Pointer getValueAtIndex(CFIndex idx) {
+            return INSTANCE.CFArrayGetValueAtIndex(this, idx);
+        }
+    }
+
+    /**
+     * A reference to an immutable {@code CFData} object.
+     */
+    class CFDataRef extends CFTypeRef {
+        public CFDataRef() {
+            super();
+        }
+
+        public CFDataRef(Pointer p) {
+            super(p);
+            if (!isTypeID(DATA_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFData. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convenience method for {@link #CFDataGetLength} on this object
+         *
+         * @return An index that specifies the number of bytes associated with this
+         *         object.
+         */
+        public CFIndex getLength() {
+            return INSTANCE.CFDataGetLength(this);
+        }
+
+        /**
+         * Convenience method for {@link #CFDataGetBytePtr} on this object
+         *
+         * @return A read-only pointer to the bytes associated with this object.
+         */
+        public Pointer getBytePtr() {
+            return INSTANCE.CFDataGetBytePtr(this);
+        }
+    }
+
+    /**
+     * A reference to an immutable {@code CFDictionary} object.
+     */
+    class CFDictionaryRef extends CFTypeRef {
+        public CFDictionaryRef() {
+            super();
+        }
+
+        public CFDictionaryRef(Pointer p) {
+            super(p);
+            if (!isTypeID(DICTIONARY_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFDictionary. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convenience method for {@link CoreFoundation#CFDictionaryGetValue} on this
+         * object.
+         *
+         * @param key
+         *            The key for which to find a match.
+         * @return The value associated with key, or {@code null} if no key-value pair
+         *         matching key exists.
+         */
+        public Pointer getValue(PointerType key) {
+            return INSTANCE.CFDictionaryGetValue(this, key);
+        }
+
+        /**
+         * Convenience method for {@link CoreFoundation#CFDictionaryGetValueIfPresent}
+         * on this object.
+         *
+         * @param key
+         *            The key for which to find a match.
+         * @param value
+         *            A pointer to memory which, on return, is filled with the
+         *            pointer-sized value if a matching key is found.
+         * @return 1 if a matching key was found, otherwise 0
+         */
+        public byte getValueIfPresent(PointerType key, PointerByReference value) {
```

Comment:
Would it make sense to convert the return of `CFDictionaryGetValueIfPresent` directly to a boolean?